File: backend/app/schemas/alert.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional, Dict, Any
from datetime import datetime
from uuid import UUID
from app.models.alert import AlertSeverity


class AlertResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    email_id: Optional[UUID] = None
    severity: AlertSeverity
    message: str
    risk_score: float
    contributing_factors: Optional[Dict[str, Any]] = None
    acknowledged: bool
    created_at: datetime


class AlertListResponse(BaseModel):
    items: List[AlertResponse]
    total: int


class AlertStatsResponse(BaseModel):
    total: int
    unacknowledged: int
    critical: int


class AlertAcknowledgeResponse(BaseModel):
    status: str
    alert_id: UUID
    acknowledged: bool

File: frontend/src/types/alert.ts
export type AlertSeverity = 'low' | 'medium' | 'high' | 'critical';

export interface AlertContributingFactors {
  title?: string;
  factors?: string[];
  recommended_action?: string;
  ioc_summary?: Array<{
    type: string;
    value: string;
    risk_score: number;
    reason?: string;
    source?: string;
  }>;
}

export interface Alert {
  id: string;
  email_id?: string;
  severity: AlertSeverity;
  message: string;
  risk_score: number;
  contributing_factors?: AlertContributingFactors | string[] | Record<string, any>;
  acknowledged: boolean;
  created_at: string;
}

export interface AlertListResponse {
  items: Alert[];
  total: number;
}

export interface AlertStats {
  total: number;
  unacknowledged: number;
  critical: number;
}

export interface AlertFilterParams {
  page?: number;
  page_size?: number;
  limit?: number;
  offset?: number;
  severity?: string;
  acknowledged?: boolean;
}

